src/commands/appSettings/AzureWebJobsStorageExecuteStep.ts

```diff
@@ -5,26 +5,34 @@
 
 import { IStorageAccountWizardContext } from '@microsoft/vscode-azext-azureutils';
 import { AzureWizardExecuteStep } from '@microsoft/vscode-azext-utils';
-import { localEmulatorConnectionString } from '../../constants';
-import { azureWebJobsStorageKey, MismatchBehavior, setLocalAppSetting } from '../../funcConfig/local.settings';
+import { ConnectionKey, ConnectionType, localStorageEmulatorConnectionString } from '../../constants';
+import { MismatchBehavior, setLocalAppSetting } from '../../funcConfig/local.settings';
 import { getStorageConnectionString } from '../../utils/azure';
 import { IAzureWebJobsStorageWizardContext } from './IAzureWebJobsStorageWizardContext';
 
 export class AzureWebJobsStorageExecuteStep<T extends IAzureWebJobsStorageWizardContext> extends AzureWizardExecuteStep<T> {
     public priority: number = 230;
 
-    public async execute(context: IAzureWebJobsStorageWizardContext): Promise<void> {
+    public constructor(private readonly _setConnectionForDeploy?: boolean) {
+        super();
+    }
+
+    public async execute(context: T): Promise<void> {
         let value: string;
-        if (context.azureWebJobsStorageType === 'emulator') {
-            value = localEmulatorConnectionString;
+        if (context.azureWebJobsStorageType === ConnectionType.NonAzure) {
```

Comment:
My reasoning was that I felt the code flows essentially boiled down to 1 of 3 paths... we are either 1) stepping the user through to create the appropriate Azure resources to connect, 2) giving the user the option to connect to some sort of resource outside of Azure instead (whether its hosted outside or locally), or 3) deferring the connection setup to a later time.

For `storage` and `netherite`, `NonAzure` does refer to emulators, whereas with `SQL`, `NonAzure` refers to any database resource outside of Azure.

Splitting `NonAzure` into `SQL` and `Emulator` means I'd have to go through and write `Omit` logic wherever I use `ConnectionTypeValues` because technically `EventHub` connections don't have a `SQL` option and there are no `Emulator` options for SQL.  In the end I chose `NonAzure` because even though the names of connections are different, I felt that their flows were very close and I didn't feel it was worth adding that extra complexity throughout just for an improvement in specificity when I could umbrella both into the same flow.

Of course I'm open to changing it, but I just wanted to explain my initial though process...  Thoughts?

Reply:
I'll take a closer re-look at this later, but I originally merged `Emulator` and `SQL` under one umbrella because it made it a lot easier to adopt my existing validation and creation flows without having to write a whole bunch of omit logic for `ConnectionTypeValues`.  Omit logic would be necessary because EventHubs can't have a `SQL` type and SQL doesn't have an `Emulator` type.  I would have to revisit all of my code flows to ensure that nothing breaks when we have those two types separated out vs under one umbrella.

Reply:
It just made adopting my existing flows a lot cleaner/simpler because I didn't have to write out different `Omit` logic for `ConnectionTypeValues` (e.g. `Netherite` can't have `SQL` and SQL doesn't have an `Emulator` option).  I remember at the time thinking it might complicate my validation logic as well but I'd really have to revisit to jog my memory on how involved that change would be

Reply:
Let's revisit this discussion next week: https://learn.microsoft.com/en-us/azure/azure-sql/database/local-dev-experience-sql-database-emulator?view=azuresql

Reply:
New issue to cover the new connection types:

https://github.com/microsoft/vscode-azurefunctions/issues/3448